### ISSUE TITLE
Guide is not clear about the API support in WebSwing

### ISSUE THREAD

**Issue by ryepesg** [zaproxy/zaproxy#3497]
I'm trying to use ZAP running on WebSwing (on top of Docker) following the instructions from: 
https://github.com/zaproxy/zaproxy/wiki/WebSwing

I even try the instructions "Accessing the API from outside of the Docker container" from:
https://github.com/zaproxy/zaproxy/wiki/Docker

To me, ZAP over WebSwing doesn't support the ZAP API, even with the API Key disabled. It would be nice if the guide tell the users it is not supported, so they don't waste time trying to make it work.

**Comment by kingthorin** [comment]
Try browsing to http://localhost:8090

**Comment by ryepesg** [comment]
This is the problem, I got "The connection was reset" with:
http://localhost:8090
http://127.0.0.1:8090
http://172.17.0.2:8090 (the Docker IP)

I tried with and without using the proxy from the browser.

**Comment by thc202** [comment]
Right, the API is no longer accessible (per latest API changes [1]). We need to change the `webswing.config` to allow all(?) addresses.

[1] https://github.com/zaproxy/zaproxy/wiki/FAQapikey

**Comment by ryepesg** [comment]
Should it work if I try with owasp/zap2docker-stable:2.**5**.0 ?

**Comment by thc202** [comment]
Yes, that one will work without further changes.

**Comment by ryepesg** [comment]
Thank you, I will use it meanwhile.

**Comment by abdoudhassi** [comment]
> Thank you, I will use it meanwhile.

ryepesg, How are you able to run the owasp/zap2docker-stable:2.5.0 with a disabled key? I'm trying 
```
docker run --rm -u zap -p 8080:8080 -p 8090:8090 -i owasp/zap2docker-stable:2.5.0 zap-webswing.sh -host 0.0.0.0 -port 8090 -config api.addrs.addr.name=.* -config api.addrs.addr.regex=true -config api.disablekey=true
```
and

```
docker run -v /path/to/webswing.config:/zap/webswing.config:rw  --rm -u zap -p 8080:8080 -p 8090:8090 -i owasp/zap2docker-stable:2.5.0 zap-webswing.sh
```

while making this edit to the webswing.config:

```
"launcherConfig" : {
        "args" : "-host 0.0.0.0 -port 8090  -config api.addrs.addr.name=.* -config api.addrs.addr.regex=true -config api.disablekey=true",
        "mainClass" : "org.zaproxy.zap.ZAP"
      }
```

I get the following error:

```
Invalid or missing API key
```



> Right, the API is no longer accessible (per latest API changes [1]). We need to change the `webswing.config` to allow all(?) addresses.
> 
> [1] https://github.com/zaproxy/zaproxy/wiki/FAQapikey

@thc202, I'm trying to edit the webswing.config file in the current zaproxy version, but I don't see where exactly. It seems that it is properly configured. However, the API UI is still resetting any connection to it.

**Comment by thc202** [comment]
Note that the path is `/zap/webswing/webswing.config` not `/zap/webswing.config`.

With the weekly image (`zap2docker-weekly`) you can change the allowed hosts through the GUI.

**Comment by ryepesg** [comment]
@abdoudhassi I gave up trying to run zap from docker after hitting this:
https://github.com/zaproxy/zaproxy/issues/3504

**Comment by james-luther** [comment]
I'm running into this issue now. I have Webswing working perfectly as well as the proxy but I cannot connect to the API. I am only receiving the following:

`curl: (52) Empty reply from server`

**Comment by thc202** [comment]
Best check the logs to see why the API request is being rejected.

**Comment by james-luther** [comment]
That was the fun part. There are no logs. When running the container with zap-webswing.sh the only thing shown in logs is 
`Using ZAP command line options: -host 0.0.0.0 -port 8080`
I was able to fix this by connecting a local volume with a custom webswing.config to /zap/webswing/webswing.config as mentioned above. :-)

**Comment by psiinon** [comment]
There will always be ZAP logs, if ZAP starts that is :)
https://www.zaproxy.org/faq/what-is-the-default-directory-that-zap-uses/

**Comment by james-luther** [comment]
> There will always be ZAP logs, if ZAP starts that is :) https://www.zaproxy.org/faq/what-is-the-default-directory-that-zap-uses/

And that was my problem. It wasn't starting 🤣

**Comment by DhanushRajGP09** [comment]
> I'm running into this issue now. I have Webswing working perfectly as well as the proxy but I cannot connect to the API. I am only receiving the following:
> 
> `curl: (52) Empty reply from server`

facing the same issue webswing works properly but while trying to use the zap api it will respond with empty reply from server if tried to hit api on postman it will throw not found error because the before the api reaching to zap server the webswing server not mapping the apis to zap and responding itself. is there any solution for this.

**Comment by psiinon** [comment]
@DhanushRajGP09 can you explain why you want to use WebSwing _and_ the ZAP API?
I've always thought of the ZAP desktop (and therefore the WebSwing interface) to be most useful for manual testing and the API for automation.
If you want to test the 2 together you can always run ZAP locally ...

**Comment by thc202** [comment]
> the webswing server not mapping the apis to zap and responding itself

There's no such mapping, you should always connect to the ZAP API directly. For the empty response, that's: https://www.zaproxy.org/faq/why-is-an-api-key-required-by-default/ or https://www.zaproxy.org/faq/how-can-i-connect-to-zap-remotely/

**Comment by DhanushRajGP09** [comment]
> @DhanushRajGP09 can you explain why you want to use WebSwing _and_ the ZAP API? I've always thought of the ZAP desktop (and therefore the WebSwing interface) to be most useful for manual testing and the API for automation. If you want to test the 2 together you can always run ZAP locally ...
Zap docker instance is running on one of my servers which i'm trying to access from my local machine on web browser on port 8080 for browser zap is running using webswing in the application there's an option in the tools > api  to enable api access from the mentioned address. with referring ro api catalogue when tried apis curl command the response is
{
    "servlet": "org.webswing.server.JavaxWebswingServlet",
    "message": "Not Found.",
    "url": "/JSON/ascan/view/status/",
    "status": "404"
} the api is not reaching to zap server. is there any configuration to be done to get response from zap server.

**Comment by thc202** [comment]
As mentioned in https://www.zaproxy.org/docs/docker/webswing/ the ZAP API will be on 8090, you should use that port when connecting to the ZAP API.

**Comment by DhanushRajGP09** [comment]
> As mentioned in https://www.zaproxy.org/docs/docker/webswing/ the ZAP API will be on 8090, you should use that port when connecting to the ZAP API.

with reference to the same doc i'm trying to access api on port 8090 but still i'm facing the same error. Please provide a docker pull command only for accessing api it will be helpful.

**Comment by psiinon** [comment]
It would be quicker if you tell us what you are using 😁 
I dont know when i'll have time to look at that, but I can review things much more quickly..

**Comment by thc202** [comment]
If you are already using 8090 then you need to check the links in https://github.com/zaproxy/zaproxy/issues/3497#issuecomment-2345916503 either give permissions to the remote address and/or provide the correct key.

**Comment by DhanushRajGP09** [comment]
> If you are already using 8090 then you need to check the links in [#3497 (comment)](https://github.com/zaproxy/zaproxy/issues/3497#issuecomment-2345916503) either give permissions to the remote address and/or provide the correct key.

docker run -u zap -p 8080:8080 -p 8090:8090 -i ghcr.io/zaproxy/zaproxy:stable zap-webswing.sh -config api.addrs.addr.name=.* -config api.addrs.addr.regex=true
Using ZAP command line options: -host 0.0.0.0 -port 8090

running the docker with the above command as mentioned and accessing the api through http://10.10.1.78:8090/JSON/ascan/view/status/ getting the response as:

{
    "servlet": "org.webswing.server.JavaxWebswingServlet",
    "message": "Not Found.",
    "url": "/JSON/ascan/view/status/",
    "status": "404"
}

**Comment by kingthorin** [comment]
If you're driving ZAP via the API why does webswing matter at all?

**Comment by thc202** [comment]
Check https://www.zaproxy.org/docs/docker/webswing/#zap-command-line-options you need to use the env var to pass the args to ZAP.
